File: models/Group.ts
import mongoose, { Schema, type Document } from "mongoose"

export interface IGroup extends Document {
  name: string
  specialty: string
  course: number
  createdAt: Date
}

const GroupSchema = new Schema<IGroup>({
  name: {
    type: String,
    required: true,
    unique: true,
  },
  specialty: {
    type: String,
    required: true,
  },
  course: {
    type: Number,
    required: true,
    min: 1,
    max: 4,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
})

export default mongoose.models.Group || mongoose.model<IGroup>("Group", GroupSchema)

File: flutter_app/lib/models/group.dart
class Group {
  final String id;
  final String name;
  final String specialty;
  final int course;
  final DateTime createdAt;

  Group({
    required this.id,
    required this.name,
    required this.specialty,
    required this.course,
    required this.createdAt,
  });

  factory Group.fromJson(Map<String, dynamic> json) {
    return Group(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      specialty: json['specialty'] ?? '',
      course: json['course'] ?? 1,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'specialty': specialty,
      'course': course,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
